provider/clusterrosaclassic/cluster_rosa_classic_resource.go

```diff
@@ -1339,201 +1373,144 @@ func (r *ClusterRosaClassicResource) Delete(ctx context.Context, request tfsdk.D
 	response.State.RemoveResource(ctx)
 }
 
-func (r *ClusterRosaClassicResource) ImportState(ctx context.Context, request tfsdk.ImportResourceStateRequest,
-	response *tfsdk.ImportResourceStateResponse) {
+func (r *ClusterRosaClassicResource) ImportState(ctx context.Context, request resource.ImportStateRequest,
+	response *resource.ImportStateResponse) {
 	tflog.Debug(ctx, "begin importstate()")
 
-	tfsdk.ResourceImportStatePassthroughID(ctx, tftypes.NewAttributePath().WithAttributeName("id"), request, response)
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), request, response)
 }
 
 // populateRosaClassicClusterState copies the data from the API object to the Terraform state.
 func populateRosaClassicClusterState(ctx context.Context, object *cmv1.Cluster, state *ClusterRosaClassicState, httpClient HttpClient) error {
-	state.ID = types.String{
-		Value: object.ID(),
-	}
-	state.ExternalID = types.String{
-		Value: object.ExternalID(),
-	}
+	state.ID = types.StringValue(object.ID())
+	state.ExternalID = types.StringValue(object.ExternalID())
 	object.API()
-	state.Name = types.String{
-		Value: object.Name(),
-	}
-	state.CloudRegion = types.String{
-		Value: object.Region().ID(),
-	}
-	state.MultiAZ = types.Bool{
-		Value: object.MultiAZ(),
-	}
-
-	state.Properties = types.Map{
-		ElemType: types.StringType,
-		Elems:    map[string]attr.Value{},
-	}
-	state.OCMProperties = types.Map{
-		ElemType: types.StringType,
-		Elems:    map[string]attr.Value{},
-	}
+	state.Name = types.StringValue(object.Name())
+	state.CloudRegion = types.StringValue(object.Region().ID())
+	state.MultiAZ = types.BoolValue(object.MultiAZ())
 	if props, ok := object.GetProperties(); ok {
+		propertiesMap := map[string]string{}
+		ocmPropertiesMap := map[string]string{}
 		for k, v := range props {
-			state.OCMProperties.Elems[k] = types.String{
-				Value: v,
-			}
+			ocmPropertiesMap[k] = v
 			if _, isDefault := OCMProperties[k]; !isDefault {
-				state.Properties.Elems[k] = types.String{
-					Value: v,
-				}
+				propertiesMap[k] = v
 			}
 		}
+		mapValue, err := common.ConvertStringMapToMapType(propertiesMap)
```

Comment:
It seems like each time we do something different with properties or i miss  something?

Reply:
IT's no the same - this function convert map[string]string to type.Map